google_photos_takeout_helper/__main__.py

```diff
@@ -174,7 +176,42 @@ def get_hash(file: Path, first_chunk_only=False, hash_algo=_hashlib.sha1):
                     hashobj.update(chunk)
         return hashobj.digest()
 
-    # PART 1: removing duplicates
+    def populate_album_map(path: Path, filter_fun=lambda f: (is_photo(f) or is_video(f))):
+        if not path.is_dir():
+            raise NotADirectoryError('populate_album_map only handles directories not files')
+
+        meta_file_exists = find_album_meta_json_file(path)
+        if meta_file_exists is None or not meta_file_exists.exists():
+            return False
+
+        # means that we are processing an album so process
+        for file in path.rglob("*"):
+            if not (file.is_file() and filter_fun(file)):
+                continue
+            file_name = file.name
+            # If it's not in the output folder
+            if not (FIXED_DIR / file.name).is_file():
+                full_hash = None
+                try:
+                    full_hash = get_hash(file, first_chunk_only=False)
+                except Exception as e:
+                    print(e)
+                    print(f"populate_album_map - couldn't get hash of {file}")
+                if full_hash is not None and full_hash in files_by_full_hash:
+                    full_hash_files = files_by_full_hash[full_hash]
+                    if len(full_hash_files) != 1:
+                        print("full_hash_files list should only be one after duplication removal, bad state")
```

Comment:
This is always printed for me :confused:

Reply:
Let me run now and check

Reply:
Nope, just standard `-i -o`...

Reply:
Okay, well, this means that this isn't getting called:
```
 # Removes all duplicates in folder
    def remove_duplicates(dir: Path):
        find_duplicates(dir, lambda f: (is_photo(f) or is_video(f)))
        nonlocal s_removed_duplicates_count

        # Now we have populated the final multimap of absolute dups, We now can attempt to find the original file
        # and remove all the other duplicates
        for files in files_by_full_hash.values():
            if len(files) < 2:
                continue  # this file size is unique, no need to spend cpu cycles on it

            s_removed_duplicates_count += len(files) - 1
            for file in files:
                # TODO reconsider which dup we delete these now that we're searching globally?
                if len(files) > 1:
                    file.unlink()
                    files.remove(file)
        return True
```

Notice, we loop through the files and only keep the last one and that's why I make that assertion. So somehow we're avoiding that call. It gets called with just `-i -o` on my end and I don't get those messages so i'm not sure how our code is different. I pulled the latest :/

```
    if not args.keep_duplicates:
        print('=====================')
        print('Removing duplicates...')
        print('=====================')
        remove_duplicates(
            dir=FIXED_DIR
        )
```

AFAIK, it would be if --keep-duplicates was true